### PR TITLE
#1749 Add active season IIDM extension management

### PR DESCRIPTION
**Pull request by lecourtoisflo**

### CODE REVIEW

File: dynawo/sources/Modeler/DataInterface/PowSyblIIDM/DYNLineInterfaceIIDM.h [hunk 1]
```diff
@@ -52,6 +54,31 @@ class LineInterfaceIIDM : public LineInterface {
    */
   explicit LineInterfaceIIDM(powsybl::iidm::Line& line);
 
+  // Delete copy constructors and assignment operator because of destruction of IIDM extensions pointers in destructor
```
Comment (rosiereflo) [inline dynawo/sources/Modeler/DataInterface/PowSyblIIDM/DYNLineInterfaceIIDM.h:57]: We usually inherit from boost:non_copyable to do this

File: dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestLineInterface.cpp [hunk 2]
```diff
@@ -173,6 +173,13 @@ TEST(DataInterfaceTest, Line) {
   li.addCurrentLimitInterface2(curLimItf2);
   ASSERT_EQ(li.getCurrentLimitInterfaces1().size(), 1);
   ASSERT_EQ(li.getCurrentLimitInterfaces2().size(), 1);
+  std::string season = "UNDEFINED";
```
Comment (rosiereflo) [inline dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestLineInterface.cpp:176]: why creating a temporary variable?

**Review by rosiereflo — CHANGES_REQUESTED** [review]

File: dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestLineInterface.cpp (hunk 2 above)
Comment (lecourtoisflo) [reply dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestLineInterface.cpp:176]: No reason, I just find it easier to understand what the UNDEFINED stands for if we have a variable saying it is "season"

File: dynawo/sources/Modeler/DataInterface/PowSyblIIDM/DYNLineInterfaceIIDM.h (hunk 1 above)
Comment (lecourtoisflo) [reply dynawo/sources/Modeler/DataInterface/PowSyblIIDM/DYNLineInterfaceIIDM.h:57]: OK

**Review by gautierbureau — APPROVED** [review]

**Review by rosiereflo — APPROVED** [review]

**Pull request merged**